pkg/util/azure/azure.go

```diff
@@ -34,6 +35,28 @@ func GetHostAlias() (string, error) {
 	return string(all), nil
 }
 
+// GetClusterName returns the name of the cluster containing the current VM
+func GetClusterName() (string, error) {
+	res, err := getResponse(metadataURL + "/metadata/instance/compute/resourceGroupName?api-version=2017-08-01&format=text")
+	if err != nil {
+		return "", fmt.Errorf("Azure ClusterName: unable to query metadata endpoint: %s", err)
+	}
+
+	defer res.Body.Close()
+	all, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return "", fmt.Errorf("error while reading response from azure metadata endpoint: %s", err)
+	}
+
+	splittedAll := strings.Split(string(all), "_")
+	if len(splittedAll) < 2 {
```

Comment:
Can we be even more restrictive on the input we accept (iirc there's more than two parts in the fixture you retrieved